### DISCUSSION TITLE
Update only deletions?

### DISCUSSION THREAD

**Discussion by Juddium** [beetbox/beets#5461] [Q&A] [answered]
Is there a way to use the update command to only look for file deletions? I have many files that I don't write metadata to, and every time I run an update because some files were deleted it goes through all those again. For the most part I only need update to detect file deletions or moves.

**Comment by jackwilsdon** [comment]
`beet update` should only be reading files that have been modified:

https://github.com/beetbox/beets/blob/65d78cb65db0faf661d8c724837aa818c4fb56bb/beets/ui/commands.py#L1660-L1667

Can you post the output of running update with `beet -vvv update`? Is it the same files being updated every time?

**Reply by Juddium** [reply]
Part of it is it's the same files being updated - changes don't seem to be actually written? The other part is I will run mbsync after update to repopulate the tags in the db.
I've attached the output of beet -vvv update.
[beetupdate.txt](https://github.com/user-attachments/files/17363173/beetupdate.txt)

**Reply by jackwilsdon** [reply, marked as answer]
Update doesn't support only matching deletions at the moment, but it might be nice to have a `--only-delete` flag to avoid making any other changes. It should be easy to add - just skipping the rest of this loop if the flag is set:

https://github.com/beetbox/beets/blob/65d78cb65db0faf661d8c724837aa818c4fb56bb/beets/ui/commands.py#L1650-L1704

As an alternative, you could use a shell script like this to only run update on deleted items:
```bash
#!/usr/bin/env bash

QUERY=()
while read -r LINE; do
	FILE_ID="${LINE%%$'\t'*}"
	FILE_PATH="${LINE#*$'\t'}"
	if ! [ -f "$FILE_PATH" ]; then
		# Add deleted items to query.
		QUERY+=("id:$FILE_ID" ",")
	fi
done < <(beet ls -f $'$id\t$path')
if [ ${#QUERY[@]} -gt 0 ]; then
	unset "QUERY[-1]" # Remove trailing comma.
	beet -vvv update --pretend "${QUERY[@]}"
else
	echo "Nothing to do."
fi
```
Remove `--pretend` once you're happy with what it's doing :grin:

**Comment by mmmeon** [comment]
`beet update -F deleted` detects deleted files.
think any file not found at the expected path is considered deleted, which would include files moved updating the database.
